### ISSUE TITLE
Possible bugs about GTO gradient calculation and Bragg-Slater radius

### ISSUE THREAD

**Issue by fate1997** [ifilot/pyqint#42]
Hi Ivo,

I recently tried to follow up on your codes to learn DFT, they are very helpful for beginners, thank you! 

I found there are some codes I am confused and they might be bugs:
1. [Here](https://github.com/ifilot/pyqint/blob/88821cbec18f82d53fd0e5d7a051e81e745aca36/pyqint/cgf.cpp#L100-L108), you use $(x-x_0)^{l-1}$ represent the gradient for the angular term, shouldn't it be $l\times (x-x_0)^{l-1}$?
2.  [Here](https://gitlab.tue.nl/ifilot/pydft/-/blob/master/pydft/atomicgrid.py?ref_type=heads#L438) in your `pydft` package, it seems that you get Bragg-Slater radius by directly passing the atom index, but since your Bragg-Slater radius list is started from hydrogen instead of atom placeholder, so the true way should be `bragg_slater.BSRADII[self.__aidx-1]`.

**Comment by ifilot** [comment]
Thank you for your kind words and bringing this error to my attention.

* The `l` term was missing in the calculation of the gradient. Furthermore, this error went undetected due to the unit test not testing for sufficiently high values for `l,m,n`. I have fixed the error and adjusted the unit tests.
* Your second point, though of course very valid, relates to a different repository. I have raised [a new issue](https://gitlab.tue.nl/ifilot/pydft/-/issues/1) in that repository where it will also be addressed in brief time.

Thank you once more for filing this issue. It is very much appreciated.
